sysstat.service.in

```diff
@@ -2,20 +2,27 @@
 # (C) 2012 Peter Schiffer (pschiffe <at> redhat.com)
 #
 # @PACKAGE_NAME@-@PACKAGE_VERSION@ systemd unit file:
-#	 Insert a dummy record in current daily data file.
-#	 This indicates that the counters have restarted from 0.
+#     1. Insert a dummy record in current daily data file when starting/stopping
+#        the unit. This indicates that the counters have restarted from 0.
+#     2. This unit is automatically stopped/started on suspend/resume,
+#        see /usr/lib/systemd/system-sleep/sysstat-sleep.sh.
+#     3. Starting this unit also starts the timer unit for periodic accounting.
 
 [Unit]
-Description=Resets System Activity Logs
+Description=System Activity Accounting
+# Start the periodic accounting process
+Requires=sysstat-collect.timer
+# but make sure we finish first, because the accounting process locks the datafile
+Before=sysstat-collect.timer sysstat-collect.service
 
 [Service]
 Type=oneshot
 RemainAfterExit=yes
 User=@CRON_OWNER@
-ExecStart=@SA_LIB_DIR@/sa1 --boot
+ExecStart=@SA_LIB_DIR@/sa1 --boot -C STARTED
```

Comment:
Even looking at monotonic counters (whether jiffies or second-since-boot counters) won't tell you if a system restart happened or not, because you cannot be absolutely sure that sadc will be called before long after reboot, long enough for every monotonic counters to have exceeded the value the had before system restart.